src/api/app/models/project.rb

```diff
@@ -234,11 +237,9 @@ def cleanup_before_destroy
     cleanup_linking_targets
 
     # deleting local devel packages
-    packages.each do |pkg|
-      if pkg.develpackage_id
-        pkg.develpackage_id = nil
-        pkg.save
-      end
+    packages.where.not(develpackage: { id: nil }) do |pkg|
+      pkg.develpackage_id = nil
+      pkg.save
```

Comment:
Not 100% sure but I think you could also just update all packages here without any performance loss. Might be worth to check

Reply:
As the packages are destroyed 3 lines down - what is the point anyway?

Reply:
yup and in the package model, we do all the trick being done there:

```
before_destroy :delete_on_backend
  before_destroy :close_requests
  before_destroy :update_project_for_product
  before_destroy :remove_linked_packages
  before_destroy :remove_devel_packages
```
https://github.com/vpereira/open-build-service/blob/refactor_project/src/api/app/models/package.rb#L87